### PR TITLE
Issue #961 flow packages from imod5

### LINKED ISSUE #961: Add classmethod to construct Package objects from iMOD5 data
The iMOD5 data model has the following peculiarities we need to deal with:
- Data for a package can be defined on different grids (extent, grid sizes). Most common is grids with different extents.
- Boundary conditions can be defined as planar grids, which need to be distributed across layers.

Good to know:
- Caching is probably less important, as all data is structured
- iMOD5 variables are defined on layers consistently amongst each other within one system

For iMOD Python this requires the following work:
- Regrid all variables to the same grid
- Allocate boundary conditions to layers if defined as planar grids (see: #628)
- Each iMOD5 river/drain system needs to become a separate package

Add classmethod to Package which can initiate a new package.
In pseudocode:

```python

class Package:
...
    @classmethod
    def from_imod5_package(cls, dis_pkg, **data):
        # target grid is idomain
        target = dis_pkg.dataset["idomain"]
        # Regrid to one consistent dataset
        dataset = cls._regrid_like(target, **data)
        return cls(**dataset_allocated)
...
```

For the Structured Discretization, the target grid is the smallest grid in the package. 
This is picked up in https://github.com/Deltares/imod-python/issues/964
Therefore the method needs to be overloaded:

```python
class StructuredDiscretization:
...
    @classmethod
    def from_imod5_package(cls, **data):
        # Find most fine-grained gridsize in data to regrid to
        # Clip to smallest extent
        target = find_smallest_grid(data)
        # Regrid to one consistent dataset
        dataset = cls._regrid_like(target, **data)

        # Add logic to prepare the dataset in the following way here:
        # IBOUND < 0 -> IDOMAIN = 0
        # TOP 3D -> TOP 2D
        # Thickness <0 -> IDOMAIN = -1

        return cls(**dataset)
```

further refinement 06-05
-----------------------------
This issue focusses on packages that are in the Flow model, and that are not  boundary conditions, and that contain grid -data (unlike the buoyancy package for example). This means sto, npf and ic. But npf was covered in a previous issue, so for this one it's sto and ic. for this issue.

### PR DESCRIPTION
**Pull request by luitjansl**
Fixes #961

# Description
implements importing from a imod5 project file for storage and initial conditions. 

# Checklist
- [X] Links to correct issue
- [ ] Update changelog, if changes affect users
- [X] PR title starts with ``Issue #nr``, e.g. ``Issue #737``
- [X] Unit tests were added
- [ ] **If feature added**: Added/extended example

### CODE REVIEW

File: imod/mf6/sto.py [hunk 1]
```diff
@@ -193,6 +200,63 @@ def render(self, directory, pkgname, globaltimes, binary):
...
+            Use this to override default regridding methods.
+
+        Returns
+        -------
+        Modflow 6 SpecificStorage package. Its specific yield is 0 and it's transient if any specific_storage
+             is larger than 0. All cells are set to inconvertible (they stay confined throughout the simulation)
+        """
+
+        data = {
+            "specific_storage": imod5_data["sto"]["storage_coefficient"],
```
[the remark this answers is not in the document]
Comment (luitjansl) [reply imod/mf6/sto.py:237]: ok, I'll switch it to storageCoefficient

File: imod/mf6/ic.py [hunk 2]
```diff
@@ -98,3 +104,50 @@ def render(self, directory, pkgname, globaltimes, binary):
...
+        regridder_settings = deepcopy(cls._regrid_method)
+        if regridder_types is not None:
+            regridder_settings.update(regridder_types)
+
+        regrid_context = RegridderWeightsCache()
+
+        new_package_data = _regrid_package_data(
+            data, target_grid, regridder_settings, regrid_context, {}
+        )
+        return InitialConditions(**new_package_data)
```
Comment (JoerivanEngelen) [inline imod/mf6/ic.py:153]: I think we can generalize this logic and put it in the parent class if we do:

```suggestion
        return cls(**new_package_data)
```

Might as well prepare for that?

File: imod/mf6/sto.py [hunk 3]
```diff
@@ -331,3 +338,60 @@ def render(self, directory, pkgname, globaltimes, binary):
...
+
+        new_package_data["convertible"] = zeros_like(
+            new_package_data["storage_coefficient"], dtype=int
+        )
+        new_package_data["transient"] = np.any(
+            new_package_data["storage_coefficient"].values > 0
+        )
+        new_package_data["specific_yield"] = None
+
+        return StorageCoefficient(**new_package_data)
```
Comment (JoerivanEngelen) [inline imod/mf6/sto.py:397]: See previous comment:

```suggestion
        return cls(**new_package_data)
```

**Review by JoerivanEngelen — APPROVED** [review]
Approved in advance, have one minor comment.

File: imod/mf6/ic.py (hunk 2 above)
Comment (luitjansl) [reply imod/mf6/ic.py]: done

File: imod/mf6/sto.py (hunk 3 above)
Comment (luitjansl) [reply imod/mf6/sto.py]: done

**Pull request merged**